product_main_supplierinfo/models/product_product.py

```diff
@@ -49,5 +49,7 @@ def _get_sellers(self):
             )
         )
         if not sellers:
-            sellers = all_sellers
+            sellers = all_sellers.filtered(lambda s: (s.product_id == self))
```

Comment:
I would suggest then:
```suggestion
            sellers = all_sellers.filtered(lambda s: (s.product_id == self))
            if not sellers:
                # Fallback on template sellers if there is none for the current variant
                sellers = all_sellers.filtered(lambda s: not s.product_id)
```

Reply:
In the same logic, added.